### PR TITLE
feat(server): optionally dump the evidences to files.

### PR DESCRIPTION
**Pull request by Arnaud-de-Grandmaison-ARM**
This allows processing / inspecting the evidence offline, for education or debug.

### CODE REVIEW

File: rust-keybroker/keybroker-server/src/main.rs [hunk 1]
```diff
@@ -99,6 +99,14 @@ async fn submit_evidence(
 
     let evidence_bytes = URL_SAFE_NO_PAD.decode(evidence_base64).unwrap(); // TODO: Error handling needed here in case of faulty base64 input
 
+    // Optionally dump the evidence to file.
+    // This can be useful for debugging or for educational purpose for example.
+    if data.args.dump_evidence_cbor {
+        let filename = format!("evidence-{challenge_id}.cbor",);
```
Comment (thomas-fossati) [inline rust-keybroker/keybroker-server/src/main.rs]: leftover comma?

File: rust-keybroker/keybroker-server/src/main.rs [hunk 2]
```diff
@@ -99,6 +99,14 @@ async fn submit_evidence(
...
+        std::fs::write(&filename, &evidence_bytes).unwrap();
```
Comment (thomas-fossati) [inline rust-keybroker/keybroker-server/src/main.rs]: exploding on this kind of error seems a bit drastic, no? :)

File: rust-keybroker/keybroker-server/src/main.rs (hunk 1 above)
Comment (Arnaud-de-Grandmaison-ARM) [reply rust-keybroker/keybroker-server/src/main.rs]: Well spotted --- clippy or the compiler did not flag it ! I'll fix it

File: rust-keybroker/keybroker-server/src/main.rs (hunk 2 above)
Comment (Arnaud-de-Grandmaison-ARM) [reply rust-keybroker/keybroker-server/src/main.rs]: Ok, I'll add an IO error kind then. This might get useful sometime in the future.

**Review by thomas-fossati — APPROVED** [review]
nice, thanks!

**Pull request merged**